File: typescript/src/api/resources/executionObjectives/client/requests/ObjectiveInput.ts

/**
 * @example
 *     {
 *         criterion: "criterion",
 *         idempotency_key: "idempotency_key",
 *         objective: "objective",
 *         run_id: "run_id",
 *         session_id: "session_id"
 *     }
 */
export interface ObjectiveInput {
    criterion: string;
    evidence_paths?: string[];
    idempotency_key: string;
    objective: string;
    run_id: string;
    session_id: string;
}

File: ruby/lib/Sikaru/execution_objectives/types/objective_input.rb
# frozen_string_literal: true

module Sikaru
  module ExecutionObjectives
    module Types
      class ObjectiveInput < Internal::Types::Model
        field :project_id, -> { String }, optional: false, nullable: false

        field :criterion, -> { String }, optional: false, nullable: false

        field :evidence_paths, -> { Internal::Types::Array[String] }, optional: true, nullable: false

        field :idempotency_key, -> { String }, optional: false, nullable: false

        field :objective, -> { String }, optional: false, nullable: false

        field :run_id, -> { String }, optional: false, nullable: false

        field :session_id, -> { String }, optional: false, nullable: false
      end
    end
  end
end
